Documentation/kubernetes-networking.md

```diff
@@ -0,0 +1,101 @@
+
+* Pod-to-Service Communication
+    * Services are implemented by assigning Virtual IPs which clients can access and are transparently proxied to the Pods grouped by that service. Requests to the Service IPs are intercepted by a kube-proxy process running on all nodes, which is then responsible for routing to the correct Pod.
+
+* External-to-Internal Communication
+    * Accessing services from outside the cluster is generally implemented by configuring external loadbalancers which target all nodes in the cluster. Once traffic arrives at a node, it is routed to the correct Service backends via the kube-proxy.
+
+See [Kubernetes Networking][kubernetes-network] for more detailed information on the Kubernetes network model and motiviation.
+
+[kubernetes-network]: http://kubernetes.io/v1.0/docs/admin/networking.htm
+
+## Port allocation
+
+The information below describes a minimum set of port allocations used by Kubernetes components. Some of these allocations will be optional depending on the deployment (e.g. if flannel is being used). Additionally, there is likely additional ports a deployer will need to open on their infrastructure (e.g. 22/ssh).
+
+Controller Node Inbound
+
+| Protocol | Port Range | Source                                    | Purpose                |
+-----------|------------|-------------------------------------------|------------------------|
+| TCP      | 443        | Worker Nodes, API Requests, and End-Users | Kubernetes API server. |
+
+Worker Node Inbound
+
+| Protocol | Port Range  | Source                         | Purpose                                                                |
+-----------|-------------|--------------------------------|------------------------------------------------------------------------|
+| TCP      | 10250       | Control Nodes                  | Worker node Kubelet healthcheck port.                                  |
+| TCP      | 30000-32767 | External Application Consumers | Default port range for [external service][external-service] ports. Typically, these ports would need to be exposed to external load-balancers, or other external consumers of the application itself. |
+| TCP      | ALL         | Worker & Control Nodes         | Intra-cluster communication (unnecessary if flannel is used)           |
+| UDP      | 8285        | Worker Nodes                   | flannel overlay network (only required if using flannel)               |
+
+etcd Node Inbound
+
+| Protocol | Port Range | Source        | Purpose                                                  |
+-----------|------------|---------------|----------------------------------------------------------|
+| TCP      | 2379       | Control Nodes | etcd server client API                                   |
+| TCP      | 2379       | Worker Nodes  | etcd server client API (only required if using flannel). |
+| TCP      | 2380       | etcd Nodes    | Used for etcd intra-cluster communication.               |
```

Comment:
2380 needs to be open to any etcd2 proxies, too, which might mean worker nodes need access to 2380